cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/SlangModellerImpl.java

```diff
@@ -63,8 +63,18 @@ private ExecutableModellingResult transformToExecutable(ParsedSlang parsedSlang,
 
     private void validateFileName(String executableName, ParsedSlang parsedSlang, ExecutableModellingResult result) {
         String fileName = parsedSlang.getName();
-        if (StringUtils.isNotEmpty(executableName) && !executableName.equals(fileName))
-            result.getErrors().add(new IllegalArgumentException("Operation/Flow " + executableName +
-                    " should be declared in a file named \"" + executableName + "." + Extension.SL.getValue() + "\""));
+        Extension fileExtension = parsedSlang.getFileExtension();
+        if (StringUtils.isNotEmpty(executableName) && !executableName.equals(fileName)) {
+            if (fileExtension == null) {
+                result.getErrors().add(new IllegalArgumentException("Operation/Flow " + executableName +
+                        " is declared in a file named \"" + fileName + "\"," +
+                        " it should be declared in a file named \"" + executableName + "\" plus a valid " +
+                        "extension(" + Extension.getExtensionValuesAsString() + ") separated by \".\""));
+            } else {
+                result.getErrors().add(new IllegalArgumentException("Operation/Flow " + executableName +
```

Comment:
If it is an invalid extension then parsedSlang.getFileExtension() will return null